CMakeLists.txt

```diff
@@ -109,16 +109,26 @@ else()
 
   # No config file for this
   if(WITH_GFLAGS)
-    find_package(gflags REQUIRED)
+    option(GFLAGS_USE_TARGET_NAMESPACE "Use gflags import target with namespace." ON)
+    find_package(gflags CONFIG)
+    if(gflags_FOUND)
+      list(APPEND THIRDPARTY_LIBS ${GFLAGS_TARGET})
```

Comment:
why not use `gflags::gflags` directly? as it's documented by https://github.com/gflags/gflags/blob/master/CMakeLists.txt#L19 as well and https://gflags.github.io/gflags/#cmake .

Reply:
It's because there're shared/static version of gflags, threaded/nonthreaded, and namespaces/plain (for old system) versions, this redirect to the actual one.
https://github.com/gflags/gflags/blob/addd749114fab4f24b7ea1e0f2f837584389e52c/cmake/config.cmake.in#L89-L109

It's also what gflags used officially: https://github.com/gflags/gflags/blob/addd749114fab4f24b7ea1e0f2f837584389e52c/cmake/config.cmake.in#L163

Reply:
i am not sure if your reply addresses my question. how is multiple builds of gflags library related not using `gflags::gflags`?

Reply:
OK I think you're right. The correct variation is also aliased to gflags::gflags.

Reply:
Oops, I think `gflags::gflags` doesn't work without `GFLAGS_USE_TARGET_NAMESPACE=ON`. `${GFLAGS_TARGET}` is still the correct one to use.

Reply:
Added GFLAGS_LIBRARIES support for 2.1